### PR TITLE
RC 2.19.3

### PR DESCRIPTION
**Pull request by aaaronlopez**
@code-crusher

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java [hunk 1]
```diff
@@ -149,16 +150,22 @@ private BranchResponse doRestfulPost(String url, JSONObject payload, int retryNu
...
-                        return new BranchResponse(getResponseString(connection.getInputStream()), responseCode);
+                        inputStream = connection.getInputStream();
                     }
+                    return new BranchResponse(getResponseString(inputStream), responseCode);
                 } catch (FileNotFoundException ex) {
                     // In case of Resource conflict getInputStream will throw FileNotFoundException. Handle it here in order to send the right status code
                     PrefHelper.Debug("BranchSDK", "A resource conflict occurred with this request " + url);
                     return new BranchResponse(null, responseCode);
+                } finally {
+                    if (inputStream != null) {
```
Comment (code-crusher) [inline Branch-SDK/src/io/branch/referral/network/BranchRemoteInterfaceUrlConnection.java]: @aaaronlopez inputStream closing should be accompanied with try-catch for the crash prevention like this:

try { 
     if (inputStream != null) 
        inputStream.close(); 
     } catch(IOException e)

**Comment by aaaronlopez** [comment]
@code-crusher what do you think now without the input stream?

**Comment by migdol** [comment]
Please restore `inputStream.close()` and wrap in a try/catch block as @code-crusher noted. Issue #600 will not be resolved without it.

**Comment by aaaronlopez** [comment]
@code-crusher @migdol adding the input stream back with try/catch. Doing a few more quality checks before release

**Comment by migdol** [comment]
Thank you

**Comment by code-crusher** [comment]
Looks good @aaaronlopez

**Pull request merged**